tests/endpoints/test_dump_qc_states.py

```diff
@@ -57,3 +60,52 @@ def test_get_qc_by_product_id(test_client: TestClient, load_data4well_retrieval)
     assert len(response_data) == 1
     assert MISSING_CHECKSUM not in response_data
     assert FIRST_GOOD_CHECKSUM in response_data
+
+
+def test_get_qc(test_client: TestClient, load_data4well_retrieval):
+
+    response = test_client.get("/products/qc")
+    assert response.status_code == 200
+    response_data = response.json()
+    assert len(response_data) == 0
+
+    response = test_client.get("/products/qc?weeks=-1")
+    assert response.status_code == 422
+
+    # Earliest test QC states are updated on 2022-02-15
+    interval = datetime.today() - datetime(year=2022, month=2, day=15)
+    num_weeks = int(interval.days / 7 + 2)
+
+    response = test_client.get(f"/products/qc?weeks={num_weeks}")
+    assert response.status_code == 200
+    response_data = response.json()
+    assert len(response_data) == 18
+    assert sum([len(l) for (id, l) in response_data.items()]) == 34
```

Comment:
In all of the subsequent `.items()` list comprehensions, you could probably just as well call `.values()`? I don't think you use `id` at all.